sql/core/src/test/java/test/org/apache/spark/sql/execution/sort/RecordBinaryComparatorSuite.java

```diff
@@ -321,4 +322,48 @@ public void testBinaryComparatorWhenOnlyTheLastColumnDiffers() throws Exception
 
     assert(compare(0, 1) < 0);
   }
+
+  @Test
+  public void testCompareLongsAsLittleEndian() {
+    long arrayOffset = 12;
```

Comment:
The test cases in this PR look problematic: the `long arrayOffset = 12;` is hard coded with implicit assumption on the underlying JVM's object layout in memory.
The object layout isn't guaranteed to be the same among different JVM, and in fact can be different even with the same JVM on the same machine but just different VM options.

It's preferable to use `Platform.LONG_ARRAY_OFFSET` as your mental model of where "0-offset" is. You should never write anything into the offset range `[0, Platform.LONG_ARRAY_OFFSET[` because there's no guarantee what's there.
e.g. on the HotSpot JVM, that's where the object header lives.

For example, on the HotSpot JVM, the object header size for `long[]` is:
- 32-bit: 16 bytes = 4 mark work + 4 klassptr + 4 array length + 4 padding
- 64-bit with Compressed Class pointer: 16 bytes = 8 mark word + 4 klassptr + 4 array length
- 64-bit without Compressed Class pointer: 24 bytes = 8 mark work + 8 klassptr + 4 array length + 4 padding.
12 is always a bad offset to use in this test case on HotSpot, regardless of which mode above it's in -- writing to this offset will corrupt the header of the `long[]` and if you happen to go into a GC right after the corruption, you'll find the VM doing weird stuff (including crashing with a SIGSEGV).